Remove client cert necessity

EdGeraghty opened the pull request:
Not everyone needs to provide a client certificate to their upstream postgresql, so I've added a flag which allows you to turn off the requirement at run time 😄

EdGeraghty committed:
Add a new `-use-client-keypair` flag, defaulted to true

EdGeraghty committed:
Only check for and use the keypair if `-use-client-keypair=true`

This is the default behaviour

EdGeraghty committed:
Update README.md to reflect update

glebarez commented:
Hello! Thank you for the contribution.
One remark: it seems like it'll be better to just make the client cert parameters non-mandatory, instead of introducing the additional one. How do you like it ?

EdGeraghty commented:
I am also happy with that as an option - I didn't want to change up too much of your code or its default behaviour 😊

EdGeraghty committed:
Roll back changes to `main.go`

After [discussion on the PR](https://github.com/glebarez/pgssl/pull/4#issuecomment-1680302316) I'l make `k` & `c` optional instead

EdGeraghty committed:
Make `clientCertPath` and `clientCertKey` optional, but throw a fatal error if only one parameter is provided

EdGeraghty committed:
Only load client certificate if provided

Otherwise - and by default - simply wrap the connection

EdGeraghty committed:
Update `README.md` to reflect dropped flag

EdGeraghty committed:
Merge branch 'glebarez:master' into remove-client-cert-necessity

glebarez reviewed (approved):
Cool! Thanks for the contribution!
